### PR TITLE
Rename `Instrumentation` and `Results Processors`

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: wa/framework/configuration/core.py [hunk 1]
```diff
@@ -891,16 +891,16 @@ class JobSpec(Configuration):
...
                            If specified, labels will be used by some result
                            processes instead of (or in addition to) the workload
-                           name. For example, the csv result processor will put
+                           name. For example, the csv output processor will put
                            the label in the "workload" column of the CSV file.
                            '''),
         ConfigurationPoint('augmentations', kind=toggle_set, merge=True,
                            aliases=["instruments", "processors", "instrumentation",
-                                    "result_processors", "augment"],
+                                    "output_processors", "augment"],
```
Comment (setrofim) [inline wa/framework/configuration/core.py]: Should keep the `"result_processors"` alias as well, for compatibility.

File: wa/framework/configuration/core.py [hunk 2]
```diff
@@ -891,16 +891,16 @@ class JobSpec(Configuration):
...
                            description='''
-                           The instruments and result processors to enable (or
+                           The instruments and output processors to enable (or
                            disabled using a ~) during this workload spec. This combines the
-                           "instrumentation" and "result_processors" from
+                           "instrumentation" and "output_processors" from
```
Comment (setrofim) [inline wa/framework/configuration/core.py]: Here, it should actually keep saying `"result_processors"` as this talks about the names of these configs in WA2.

File: wa/framework/configuration/core.py (hunk 1 above)
Comment (marcbonnici) [reply wa/framework/configuration/core.py]: Fixed

File: wa/framework/configuration/core.py (hunk 2 above)
Comment (marcbonnici) [reply wa/framework/configuration/core.py]: Fixed

**Pull request merged**